packages/core/src/html/NodeProcessor.ts

```diff
@@ -343,8 +395,8 @@ class NodeProcessor {
         const mainHtmlNodes = dom.map((d) => {
           let processed;
           try {
-            processed = this._process(d, context);
-          } catch (err) {
+            processed = this.traverse(d, context);
```

Comment:
`this.traverse` can return a `DomElement | DomElement[]`

Reply:
Nice catch, changed to just `DomElement`. Also resolves the other issue you pointed out with `_trimNodes`